### PR TITLE
added support for springy actions

### PR DESCRIPTION
**Pull request by jvisenti**
for issue #1

### CODE REVIEW

File: RZViewActions/UIView+RZViewActions.h [hunk 1]
```diff
@@ -87,6 +87,15 @@ typedef void (^RZViewActionCompletion)(BOOL finished);
 + (instancetype)action:(RZViewActionBlock)action withOptions:(UIViewAnimationOptions)options duration:(NSTimeInterval)duration;
 
 /**
+ *  Returns a new springy action.
+ *  The action block is used as a UIView animation block, and therefore must not be nil.
+ *
+ *  @note The action block is copied, so make sure to use weak references to objects that may retain ownnership of the action.
```
Comment (ZevEisenberg) [inline RZViewActions/UIView+RZViewActions.h]: ownnership

File: RZViewActions/UIView+RZViewActions.m [hunk 2]
```diff
@@ -128,6 +142,17 @@ - (void)_runWithCompletion:(RZViewActionCompletion)completion
 
 @end
 
+#pragma mark - RZViewSpringAction
+
+@implementation RZViewSpringAction
+
+- (void)_runWithCompletion:(RZViewActionCompletion)completion
```
Comment (ZevEisenberg) [inline RZViewActions/UIView+RZViewActions.m]: Underscore-prefixed methods are reserved by Apple.
justin is wrong

Comment (jvisenti) [inline RZViewActions/UIView+RZViewActions.m]: Lame. Is it stated anywhere that they're reserved by Apple, or is that just the consensus?

Comment (jkaufman) [inline RZViewActions/UIView+RZViewActions.m]: I knew this would haunt me, but man, @ZevEisenberg, this is necromancy.

Comment (jkaufman) [inline RZViewActions/UIView+RZViewActions.m]: Yes, @jvisenti, it's called out in the ["Naming Methods" section of Coding Guidelines for Cocoa](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingMethods.html)

```
Don’t use the underscore character as a prefix for your private methods. Apple reserves this convention.
```

**Comment by jvisenti** [comment]
@ZevEisenberg Just made those changes. Are you ok with merging this?

File: RZViewActions/UIView+RZViewActions.m (hunk 2 above)
Comment (jvisenti) [inline RZViewActions/UIView+RZViewActions.m]: @jkaufman Yeah I read those over lunch and was saddened. They suggest prefixing private methods as an alternative, which imo is better than doing nothing.

File: RZViewActions/UIView+RZViewActions.m [hunk 3]
```diff
@@ -90,9 +97,16 @@ + (instancetype)action:(RZViewActionBlock)action withDuration:(NSTimeInterval)du
 
 + (instancetype)action:(RZViewActionBlock)action withOptions:(UIViewAnimationOptions)options duration:(NSTimeInterval)duration
 {
-    NSParameterAssert(action);
+    return [[RZViewAction alloc] initWithBlock:action options:options duration:duration];
+}
+
++ (instancetype)springAction:(RZViewActionBlock)action withDamping:(CGFloat)dampingRatio initialVelocity:(CGFloat)velocity options:(UIViewAnimationOptions)options duration:(NSTimeInterval)duration
```
Comment (ZevEisenberg) [inline RZViewActions/UIView+RZViewActions.m]: Shouldn’t this return `RZViewSpringAction`? And that raises the question of whether this method should perhaps be on `RZViewSpringAction` instead of `RZViewAction`?

File: RZViewActions/UIView+RZViewActions.m [hunk 4]
```diff
@@ -90,9 +97,16 @@ + (instancetype)action:(RZViewActionBlock)action withDuration:(NSTimeInterval)du
 
 + (instancetype)action:(RZViewActionBlock)action withOptions:(UIViewAnimationOptions)options duration:(NSTimeInterval)duration
 {
-    NSParameterAssert(action);
+    return [[RZViewAction alloc] initWithBlock:action options:options duration:duration];
```
Comment (ZevEisenberg) [inline RZViewActions/UIView+RZViewActions.m:89]: Should use `[[self class] alloc]`

File: RZViewActions/UIView+RZViewActions.m (hunk 3 above)
Comment (jvisenti) [inline RZViewActions/UIView+RZViewActions.m]: `RZViewSpringAction` is a private class. Maybe all of these methods should return `RZViewAction *` instead of `instancetype`?

File: RZViewActions/UIView+RZViewActions.m (hunk 4 above)
Comment (jvisenti) [inline RZViewActions/UIView+RZViewActions.m:89]: Wouldn't it just be `[self alloc]`? That's what was here before, and I changed it for a reason I can't remember.

Comment (ZevEisenberg) [inline RZViewActions/UIView+RZViewActions.m:89]: Oops, yes. Class method.

Comment (jvisenti) [inline RZViewActions/UIView+RZViewActions.m:89]: I'm going to leave it as `[RZViewAction alloc]` and change all the `instancetype`s to `RZViewAction *` since that is more correct.

**Comment by jvisenti** [comment]
@Raizlabs/maintainers-ios I'd like to merge this today. If anyone has comments, please make them before this afternoon

File: RZViewActions/UIView+RZViewActions.m [hunk 5]
```diff
@@ -72,30 +77,28 @@ + (void)rz_runAction:(RZViewAction *)action withCompletion:(RZViewActionCompleti
...
-    }
-    return self;
+    return [self action:action withOptions:kNilOptions duration:duration];
 }
 
-+ (instancetype)action:(RZViewActionBlock)action withDuration:(NSTimeInterval)duration
++ (RZViewAction *)action:(RZViewActionBlock)action withOptions:(UIViewAnimationOptions)options duration:(NSTimeInterval)duration
 {
-    return [self action:action withOptions:kNilOptions duration:duration];
+    return [[RZViewAction alloc] initWithBlock:action options:options duration:duration];
```
Comment (alexrrouse) [inline RZViewActions/UIView+RZViewActions.m:89]: Why was this changed from `self alloc`?  if this is subclassable, we are better off using 'self'

**Comment by alexrrouse** [comment]
@jvisenti just one question and you have my blessing.

File: RZViewActions/UIView+RZViewActions.m (hunk 5 above)
Comment (jvisenti) [inline RZViewActions/UIView+RZViewActions.m:89]: It's not really subclass able. It's more like a class cluster in that there are class methods for creating various `RZViewActions`, but the implementation will in fact return the appropriate private subclass. I think `instancetype` was actually misleading originally, and could be misused.
